### PR TITLE
remove redundant field sseDiscoveryOptions incrementalSSEDiscoveryOpt…

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
…ions mcpOptions

1. The pilot` Server.{sseDiscoveryOptions,incrementalSSEDiscoveryOptions,mcpOptions}` are redundant, remove them

2. Simplify cleanup function calling

### CODE REVIEW

File: pilot/pkg/bootstrap/configcontroller.go [hunk 1]
```diff
@@ -125,7 +125,8 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 	var conns []*grpc.ClientConn
 	var configStores []model.ConfigStoreCache
 
-	s.mcpOptions = &mcp.Options{
+	mcpOptions := &mcp.Options{
+		ClusterID:    s.clusterID,
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/configcontroller.go]: why change this?

File: pilot/pkg/serviceregistry/mcp/controller.go [hunk 2]
```diff
@@ -76,10 +76,6 @@ func NewController(options *Options) Controller {
 	descriptorsByMessageName := make(map[string]schema.Instance, len(schemas.Istio))
 	synced := make(map[string]bool)
 	for _, descriptor := range schemas.Istio {
-		// do not register SSE as there is a dedicated controller
-		if descriptor.Collection == schemas.SyntheticServiceEntry.Collection {
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/mcp/controller.go]: why change this?

File: pilot/pkg/bootstrap/configcontroller.go [hunk 3]
```diff
@@ -285,21 +286,18 @@ func mcpSecurityOptions(ctx context.Context, cancel context.CancelFunc, configSo
...
 	conn *grpc.ClientConn,
 	reporter monitoring.Reporter,
 	clients *[]*sink.Client,
 	configStores *[]model.ConfigStoreCache) {
 	clientNodeID := ""
-	collections := make([]sink.CollectionOptions, 0, len(schemas.Istio)-1)
+	collections := make([]sink.CollectionOptions, 0, len(schemas.Istio))
 	for _, c := range schemas.Istio {
-		// do not register SSEs for this controller as there is a dedicated controller
-		if c.Collection == schemas.SyntheticServiceEntry.Collection {
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/configcontroller.go]: ` schemas.Istio` does not include `SyntheticServiceEntry` anymore.

File: pilot/pkg/serviceregistry/mcp/controller.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/mcp/controller.go]: schemas.Istio does not include SyntheticServiceEntry anymore.

File: pilot/pkg/bootstrap/configcontroller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/configcontroller.go]: We should pass the clusterID

Comment (howardjohn) [reply pilot/pkg/bootstrap/configcontroller.go]: why? I don't feel comfortable adding random things without understanding what is being fixed

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/configcontroller.go]: I think we missed setting this value, by default it is `Kubernetes`

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/configcontroller.go]: Ah, just look again, it seems this field is not used by mcp controller, to clean code i should remove it as well.

**Comment by hzxuzhonghu** [comment]
Simplified `cancel` calling, in case we add more logic and miss calling cancel in future

**Comment by hzxuzhonghu** [comment]
`TestQuotaCache` flakes too often

/retest

**Pull request merged**